api/client/client.go

```diff
@@ -96,11 +96,19 @@ func (c *Client) NewStore(ctx context.Context) (string, error) {
 }
 
 // RegisterSchema registers a new model shecma
-func (c *Client) RegisterSchema(ctx context.Context, storeID, name, schema string) error {
+func (c *Client) RegisterSchema(ctx context.Context, storeID, name, schema string, indexes []*store.IndexConfig) error {
```

Comment:
Related to this and the use of protos below, it would be nice to consider a higher level object that contains the JSON Schema and indexes. Ideally, this object would allow you so set indexes alongside the type (mongo style), among other (future) things like pre-commit state validators, which we discussed with OpenBazaar.

Reply:
I like both of these ideas! Should we try to include them here in this PR, or create issues?

Reply:
For now, my vote would be to make them variadic. Once other stuff is needed in the schema configuration, we could move to something that doesn't require the user to supply the JSON schema themselves... at which point pulling the indexes into that object would make sense.

Reply:
Perfect. I'll add that then... easy change.